rmw_zenoh_cpp/src/detail/graph_cache.cpp

```diff
@@ -1315,4 +1329,20 @@ std::unique_ptr<rmw_zenoh_event_status_t> GraphCache::take_event_status(
   status_to_take.current_count_change = 0;
   return result;
 }
+
+///=============================================================================
+void GraphCache::set_querying_subscriber_callback(
+  const std::string & keyexpr,
+  QueryingSubscriberCallback cb)
+{
+  auto cb_it = querying_subs_cbs_.find(keyexpr);
+  if (cb_it == querying_subs_cbs_.end()) {
+    std::vector<QueryingSubscriberCallback> cbs = {};
+    cbs.push_back(std::move(cb));
+    querying_subs_cbs_[keyexpr] = std::move(cbs);
+    return;
+  }
+  cb_it->second.push_back(std::move(cb));
```

Comment:
I think we can tighten this up:

```suggestion
  auto cb_it = querying_subs_cbs_.find(keyexpr);
  if (cb_it == querying_subs_cbs_.end()) {
    querying_subs_cbs_[keyexpr] = std::move(std::vector<QueryingSubscriberCallback>{});
    cb_it = querying_subs_cbs_.find(keyexpr);
  }
  cb_it->second.push_back(std::move(cb));
```